zio-http/src/main/scala/zio/http/netty/client/NettyConnectionPool.scala

```diff
@@ -114,8 +108,107 @@ object NettyConnectionPool {
       _             <- NettyFutureExecutor.executed(channelFuture)
       result        <- ZIO.attempt(channelFuture.channel())
     } yield result
+
+    for {
+      resolvedHosts <- dnsResolver.resolve(location.host)
+      sortedHosts    = sortHostsHappyEyeballs(resolvedHosts)
+      createChannels = sortedHosts.map(createChannel).toList
```

Comment:
If I replace this line with 

`sortedHosts.map(createChannel).map(_.debug("create channel")).toList`

the create channel log comes out as a success, but the function still fails to terminate.